src/ExpandablePanel/index.js

```diff
@@ -0,0 +1,222 @@
+// @flow
+import React, { Component } from "react";
+import filter from "lodash/filter";
+import clone from "lodash/clone";
+import classNames from "classnames";
+import withStyles from "react-jss";
+import Paper from "../Paper";
+import CaretRight from "../Icon/CaretRight";
+
+type Props = {
+  children: any,
+  classes: Object,
+  className: string,
+  expanded: boolean,
+  innerClassName: string,
+  theme: Object,
+  title: string,
+};
+
+type State = {
+  expanded: boolean,
+  isExpandablePanels: boolean,
+  panels: boolean[],
+};
+
+class ExpandablePanel extends Component<Props, State> {
+  constructor(props) {
+    super(props);
+    let panels = [];
+    const isExpandablePanels = Array.isArray(this.props.children)
+      ? filter(React.Children.toArray(this.props.children), (item) => {
+          return item.type.displayName === "Jss(ExpandablePanel)";
+        }).length > 0
+      : false;
+    if (isExpandablePanels) {
```

Comment:
React Components caring about or parsing their children is usually a red flag.

Components that allow `props.children` should be able to be nested recursively without side effects. Instead of worrying about children here let's refactor this component to do the following:

1. Each expandable panel has an internal state that cares about `isExpanded`
2. There is a toggle that doesn't `togglePanel` (since we aren't adding a panel to the vDOM), but either we call it `toggleExpand` or `toggleCollapse` to better indicate what's happening in the React Component. The variable name should be based on the initial state. If it's expanded by default we should call it `toggleCollapse`, if it's collapsed by default we should call it `toggleExpand` 
3. Clicking the panel header triggers `toggle<Collapse|Expand>` depending on what we name it. 
4. The panel header should not be a button, it should be a div
5. In the render function in the body of the panel, we pass `props.children` (we can remove the check for the `type` of props.children)

That way: 

- the Expandable Panel can only be toggled by clicking the header
- it doesn't matter how many children Expandable Panels the user adds because each has an onClick handler that, by design, cannot interfere with other components because the onClick handler only affects the internal toggled state for that one component. 

How React Material UI does it (for inspiration): 
- https://codesandbox.io/s/rj2zpz2nr4 
- https://codesandbox.io/s/820ryp2y8

Reply:
👏  This is a wonderful explanation Jason, thank you!

Only thing I have to add is super minor but regarding #4– anything that is clickable (the panel header) *should* be a button– attaching `onClick` to `div`s is naughty. (and I've probably done it in other places but technically shouldn't've) 😅